src/extension.ts

```diff
@@ -255,6 +257,10 @@ export async function activate(context: vscode.ExtensionContext) {
 
         recordToolkitInitialization(activationStartedOn, getLogger())
 
+        const credWatcher = createCredentialsFileWatcher(getCredentialsFilename(), toolkitSettings, loginManager)
+        const configWatcher = createCredentialsFileWatcher(getConfigFilename(), toolkitSettings, loginManager)
+        context.subscriptions.push({ dispose: () => credWatcher.close() }, { dispose: () => configWatcher.close() })
```

Comment:
thanks for moving most of the logic into `credentialsFile`, that is a better place for it. Let's go further and move all of this logic into `credentialsFile.initWatch(context, toolkitSettings, loginManager)`